Decompiler/Program.cs

```diff
@@ -654,7 +654,21 @@ private static void DumpVPK(Package package, string type, string newType)
                     {
                         using (var memory = new MemoryStream(output))
                         {
-                            resource.Read(memory);
+                            try
+                            {
+                                resource.Read(memory);
+                            }
+                            catch (Exception)
+                            {
+                                lock (ConsoleWriterLock)
+                                {
+                                    Console.ForegroundColor = ConsoleColor.DarkRed;
+                                    Console.WriteLine("\tDecompiler for resource type " + type + " not implemented, extracting as-is");
```

Comment:
You should print the actual exception instead of assuming it's not implemented.

Reply:
Well, it's descriptive enough (and having a large exception message there won't help much).

Reply:
Would this work?  
```c#
Console.WriteLine("\t" + e.Message + " on resource type " + type + ", extracting as-is");
```